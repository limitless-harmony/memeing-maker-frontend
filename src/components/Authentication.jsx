import React, { Component } from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import {
  FacebookColored,
  GoogleColored,
  LinkedinColored,
} from 'components/Icons';

class Authentication extends Component {
  handleLogin = strategy => {
    console.log(strategy);
  };

  loginWithGoogle = () => {
    return this.handleLogin('google');
  };

  loginWithFacebook = () => {
    return this.handleLogin('facebook');
  };

  loginWithLinkedIn = () => {
    return this.handleLogin('linkedIn');
  };

  render() {
    return (
      <StyledAuthentication>
        <TextSection>Sign up or log in</TextSection>
        <TextSection>to create, share, & discover</TextSection>
        <TextSection> our full library of meaningful memes.</TextSection>
        <ButtonSection>
          <Button onClick={this.loginWithGoogle}>
            <GoogleColored />
          </Button>
          <Button onClick={this.loginWithFacebook}>
            <FacebookColored />
          </Button>
          <Button onClick={this.loginWithLinkedIn}>
            <LinkedinColored />
          </Button>
        </ButtonSection>
      </StyledAuthentication>
    );
  }
}

const StyledAuthentication = styled.div`
  margin: ${calculateRem(-70)} auto auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  text-align: center;
`;

const TextSection = styled.div`
  align-items: center;
  font-size: ${calculateRem(18)};
`;

const ButtonSection = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: ${calculateRem(20)} auto;
`;

const Button = styled.div`
  margin: 0 ${calculateRem(12)};
`;

export default Authentication;
