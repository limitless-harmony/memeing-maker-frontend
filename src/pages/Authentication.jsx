import React, { Component } from 'react';
import { connect } from 'react-redux';
import { parse } from 'query-string';
import { bindActionCreators } from 'redux';

import { login } from 'actions/auth';
import Loading from 'components/Loading';

export class Authentication extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    const { match, location } = props;
    const { code } = parse(location.search);
    const { provider } = match.params;
    const { REACT_APP_API_URL } = process.env;
    if (
      !code &&
      (provider === 'facebook' ||
        provider === 'google' ||
        provider === 'linkedin')
    ) {
      window.location = `${REACT_APP_API_URL}/auth/${provider}`;
    }
  }

  componentDidMount = async () => {
    const {
      actions,
      match,
      previousPath,
      location,
      history,
      authenticated,
    } = this.props;
    const { code } = parse(location.search);
    const { provider } = match.params;
    if (!code) return null;
    const path = previousPath || '/';
    if (authenticated) return history.push(path);
    await actions.login(code, provider);
    return history.push(path);
  };

  render() {
    return <Loading />;
  }
}

const mapStateToProps = state => ({
  previousPath: state.auth.previous,
  authenticated: state.auth.authenticated,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ login }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Authentication);
