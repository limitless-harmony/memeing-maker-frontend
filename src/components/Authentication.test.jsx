import React from 'react';
import { shallow } from 'enzyme';

import Authentication from 'components/Authentication';

const mockFunction = jest.fn();
describe('Testing the Authentication component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Authentication />);
    expect(wrapper).toMatchSnapshot();
  });
  it('it calls the handleLogin method when an icon is clicked', () => {
    const wrapper = shallow(<Authentication />);
    const instance = wrapper.instance();
    instance.handleLogin = mockFunction;
    instance.loginWithGoogle();
    instance.loginWithFacebook();
    instance.loginWithLinkedIn();
    expect(mockFunction).toHaveBeenCalledTimes(3);
  });
});
