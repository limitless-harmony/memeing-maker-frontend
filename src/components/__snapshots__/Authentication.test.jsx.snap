// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Authentication component renders without crashing 1`] = `
<styled.div>
  <styled.div>
    Sign up or log in
  </styled.div>
  <styled.div>
    to create, share, & discover
  </styled.div>
  <styled.div>
     our full library of meaningful memes.
  </styled.div>
  <styled.div>
    <styled.div
      onClick={[Function]}
    >
      <ForwardRef />
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <ForwardRef />
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <ForwardRef />
    </styled.div>
  </styled.div>
</styled.div>
`;
